typescript/sdk/src/ism/EvmIsmModule.ts

```diff
@@ -86,4 +186,610 @@ export class EvmIsmModule extends HyperlaneModule<
       config,
     });
   }
+
+  protected async updateRoutingIsm(params: {
+    destination: ChainName;
+    config: RoutingIsmConfig;
+    origin?: ChainName;
+    mailbox?: Address;
+    existingIsmAddress: Address;
+    logger: Logger;
+  }): Promise<IRoutingIsm> {
+    const { destination, config, mailbox, existingIsmAddress, logger } = params;
+
+    // 1. Filter for known domains
+    const { domains, safeConfigDomains } = EvmIsmModule.filterRoutingIsmDomains(
+      { config, multiProvider: this.multiProvider },
+    );
+    config.domains = domains;
```

Comment:
imo in future we should adapt the configs themselves to accommodate non-evm keys